src/ballet/block/test_microblock.c

```diff
@@ -0,0 +1,148 @@
+#include "fd_microblock.h"
+
+#include "../../util/sanitize/fd_sanitize.h"
+
+#include <stddef.h>
+
+/* Data layout checks */
+
+FD_STATIC_ASSERT( offsetof( fd_microblock_hdr_t, hash_cnt )==0x00UL, alignment );
+FD_STATIC_ASSERT( offsetof( fd_microblock_hdr_t, hash     )==0x08UL, alignment );
+FD_STATIC_ASSERT( offsetof( fd_microblock_hdr_t, txn_cnt  )==0x28UL, alignment );
+FD_STATIC_ASSERT( sizeof  ( fd_microblock_hdr_t           )==0x30UL, alignment );
+FD_STATIC_ASSERT( alignof ( fd_microblock_hdr_t           )==0x01UL, alignment );
+
+FD_STATIC_ASSERT( alignof(fd_microblock_t)== 64UL, alignment );
+FD_STATIC_ASSERT( sizeof (fd_microblock_t)==128UL, alignment );
+
+void
+test_microblock() {
+  /* Ensure footprint is multiple of align */
+
+  for( ulong i=0UL; i<64UL; i++ ) {
+    FD_TEST( ( fd_microblock_footprint( i ) % alignof(fd_microblock_t) )==0UL );
+  }
+
+  /* Test overflowing txn_max_cnt */
+
+  FD_TEST( fd_microblock_footprint( 5167155202719762UL )==0xfffffffffffffbc0UL );
+  FD_TEST( fd_microblock_footprint( 5167155202719763UL )==0UL  ); /* invalid */
+  FD_TEST( fd_microblock_new      ( (void *)64UL, 5167155202719763UL )==NULL ); /* fake shmem, invalid footprint */
+
+  /* Test failure cases for fd_microblock_new */
+
+  FD_TEST( fd_microblock_new( NULL,        16UL )==NULL ); /* null shmem */
+  FD_TEST( fd_microblock_new( (void *)1UL, 16UL )==NULL ); /* misaligned shmem */
+
+  /* Test failure cases for fd_microblock_join */
+
+  FD_TEST( fd_microblock_join( NULL        )==NULL ); /* null shblock */
+  FD_TEST( fd_microblock_join( (void *)1UL )==NULL ); /* misaligned shblock */
+
+  /* Test microblock creation */
+
+  ulong txn_max_cnt = 16UL;
+  ulong footprint = fd_microblock_footprint( txn_max_cnt );
+  FD_TEST( footprint );
+
+  void * shmem = fd_alloca( FD_MICROBLOCK_ALIGN, footprint );
+  FD_TEST( shmem );
+
+  void * shblock = fd_microblock_new( shmem, txn_max_cnt );
+
+  fd_microblock_t * block = fd_microblock_join( shblock );
+  FD_TEST( block );
+
+  /* Test bad magic value */
+
+  block->magic++;
+  FD_TEST( fd_microblock_join( shblock )==NULL );
+  block->magic--;
+
+  /* Test microblock destruction */
+
+  FD_TEST( fd_microblock_leave( NULL  )==NULL    ); /* null block */
+  FD_TEST( fd_microblock_leave( block )==shblock ); /* ok */
+
+  FD_TEST( fd_microblock_delete( NULL        )==NULL ); /* null shblock */
+  FD_TEST( fd_microblock_delete( (void *)1UL )==NULL ); /* misaligned shblock */
+
+  /* Test bad magic value.
+     Note that at this point our `block` pointer is dangling. */
+  block->magic++;
+  FD_TEST( fd_microblock_delete( shblock )==NULL );
+  block->magic--;
+
+  FD_TEST( fd_microblock_delete( shblock )==shmem );
+}
+
+/* A serialized batch of entries.
```

Comment:
`entries` and `entry` occur a few times in this file. Can you rename them all to microblocks?